src/watcher/store.ts

```diff
@@ -74,35 +77,37 @@ export class AppointmentStore extends StartStopService implements IAppointmentSt
      * Returns true if the supplied item was adde or updated in the store.
      * @param appointment
      */
-    public async addOrUpdateByStateLocator(appointment: IEthereumAppointment): Promise<boolean> {
-        const currentAppointment = this.appointmentsByStateLocator[appointment.getStateLocator()];
-        // is there a current appointment
-        if (currentAppointment) {
-            if (currentAppointment.getStateNonce() >= appointment.getStateNonce()) {
-                logger.info(
-                    appointment.formatLog(
-                        `Nonce ${appointment.getStateNonce()} is lower than current appointment ${
-                            currentAppointment.id
-                        } nonce ${currentAppointment.getStateNonce()}`
-                    )
-                );
-
-                return false;
-            } else {
-                // remove the old appointment
-                delete this.appointmentsById[currentAppointment.id];
+    public addOrUpdateByStateLocator(appointment: IEthereumAppointment): Promise<boolean> {
+        return this.lockManager.withLock(appointment.getStateLocator(), async () => {
```

Comment:
I think you should include a detailed comment here on why and when the state locator needs to be locked. Also on why the appointment id does not need to be locked.

Reply:
Done in 736c3f65ce697ca696ab734d873fbc468b4b0590.